pkg/controller/test-images-distributor/test_images_distributor.go

```diff
@@ -291,6 +330,22 @@ func (r *reconciler) reconcile(ctx context.Context, req reconcile.Request, log *
 	*log = *log.WithField("cluster", cluster).WithField("namespace", decoded.Namespace).WithField("name", decoded.Name)
 	log.Info("Starting reconciliation")
 
+	// One of the following is allowed:
+	// - multiarch namespaces to distribute on the proper non-amd64 clusters (ex.: ci-arm64 on arm01)
+	// or
+	// - non-multiarch namespaces to distribute on any amd64 clusters (ex.: ci on build03)
+	if isMultiarchNamespace(decoded.Namespace) {
+		if !isNamespaceAllowedOnCluster(decoded.Namespace, cluster) {
+			log.Debug("multiarch imageStreamTag not allowed on cluster")
+			return nil
+		}
+	} else {
```

Comment:
you mean `else if` here?

Reply:
Yes, initially I wrote only one `if` condition but it was unreadable. This is the easiest form to understand, if we agree on that I can refine it a little bit.